src/global/vars.h

```diff
@@ -147,6 +148,7 @@ extern REQUEST_INFO LoadSaveGameRequester;
 
 extern int32_t HealthBarTimer;
 extern int16_t StoredLaraHealth;
+extern int16_t LevelStartLaraHealth;
```

Comment:
I don't think we need two variables to signify Lara's health – just one should've been enough?

Reply:
Well `StoredLaraHealth` gets updated every loop. So what was happening was, when Lara died, she would restart the level and instantly die on respawn since she had no health. Also, if  a restart level was ever implemented that could happen at any time, she would restart with her current health, not the health she entered the level with.